### PR TITLE
Remove positive only classifications

### PR DESCRIPTION
**Pull request by hudson-newey**
# Remove positive only classifications

### Features

- Removes optional classifications
- Removes the color fill that is applied to used decision buttons

### Bug Fixes

- Fixes a bug where the fullscreen-exit icon would not show in the settings component

## Visual Changes

image

_New classification task groups_

image

_New classification task group with a long tag title_

image

_Text for keyboard shorcuts displayed in help prompt_

## Documentation Examples

<https://66d80ca77a086955056280a5--oe-web-components.netlify.app/examples/verification/classification/>

## Related Issues

Fixes: #135 (Remove positive only classifications)
Fixes: #160 (Add a pill of color to decision buttons indicating the color assigned to the class)
Fixes: #122 (Improve semantic meaning of a decision button with a highlight color)

## Final Checklist

- [x] All commits messages are semver compliant
- [x] Added relevant unit tests for new functionality
- [x] Updated existing unit tests to reflect changes
- [x] Code style is consistent with the project guidelines
- [x] Documentation is updated to reflect the changes (if applicable)
- [x] Link issues related to the PR
- [x] Assign labels if you have permission
- [x] Assign reviewers if you have permission
- [x] Ensure that CI is passing
- [x] Ensure that `pnpm lint` runs without any errors
- [x] Ensure that `pnpm test` runs without any errors

### CODE REVIEW

**Comment by atruskie** [comment]
> image

There seems to be redundant text in the descriptions

**Comment by atruskie** [comment]
Missing visual example for verification:

image

File: src/components/decision/classification/classification.ts [hunk 1]
```diff
@@ -1,28 +1,185 @@
...
+    const color = this.injector.colorService(decisionModel);
+
+    return html`
+      <button
+        id="${decision}-decision-button"
+        class="decision-button ${buttonClasses}"
+        part="${decision}-decision-button"
+        title="Shortcut: ${shortcut}"
+        style="--ripple-color: var(${color})"
+        aria-disabled="${this.disabled}"
```
Comment (atruskie) [inline src/components/decision/classification/classification.ts:155]: are we missing an ordinary disabled button?

File: src/components/decision/classification/classification.ts [hunk 2]
```diff
@@ -1,28 +1,185 @@
...
+
+  public override render() {
+    const buttonOptions = [DecisionOptions.TRUE, DecisionOptions.FALSE];
+
+    // prettier-ignore
+    return html`
+      <div class="decision-group">
+        <div class="decision-group-title">
+          ${this.tag.text ?? this.tag}
+          <slot></slot>
```
Comment (atruskie) [inline src/components/decision/classification/classification.ts:174]: todo for normalizing the tag model. Should not need to fallback to outputting `this.tag`.

We should also move the text into the slot, so if the slot is customized by the user it overrides the text.

File: src/components/decision/css/style.css [hunk 3]
```diff
@@ -27,24 +27,9 @@
...
 
-/*
-  I declare the --decision-color variable on the decision button class not
-  on the id selector because I need the specificity to be 0,1,0 so that I can
-  override the variable with a decision class
- */
-.decision-button {
-  --decision-color: none;
+.decision-color-pill {
+  padding: 0.2em;
```
Comment (atruskie) [inline src/components/decision/css/style.css:33]: don't pad vertically

File: src/components/decision/verification/verification.ts [hunk 4]
```diff
@@ -1,21 +1,66 @@
...
+  public shortcut = "";
+
+  @query("#decision-button")
+  private decisionButton!: HTMLButtonElement;
+
+  public override shortcutKeys(): KeyboardShortcut[] {
+    let description = `Apply a ${this.verified} verification`;
+    if (this.additionalTags.length) {
+      const additionalTagText = this.additionalTags.map((tag) => tag.text).join(", ");
+      description += ` and classifications: ${additionalTagText}`;
```
Comment (atruskie) [inline src/components/decision/verification/verification.ts:45]:
```suggestion
      description += ` and also these tags: ${additionalTagText}`;
```

File: src/components/decision/verification/verification.ts [hunk 5]
```diff
@@ -26,4 +71,45 @@ export class VerificationComponent extends DecisionComponent {
...
+    const color = this.injector.colorService(verificationModel);
+
+    return html`
+      <button
+        id="decision-button"
+        class="oe-btn-primary decision-button ${buttonClasses}"
+        part="decision-button"
+        title="Shortcut: ${this.shortcut}"
+        style="--ripple-color: var(${color})"
+        aria-disabled="${this.disabled}"
```
Comment (atruskie) [inline src/components/decision/verification/verification.ts:97]: disabled attribute?

File: src/components/decision/verification/verification.ts [hunk 6]
```diff
@@ -26,4 +71,45 @@ export class VerificationComponent extends DecisionComponent {
...
+        @click="${() => this.emitDecision(decisionModels)}"
+      >
+        <span class="oe-pill decision-color-pill" style="background-color: var(${color})"></span>
+
+        <div class="button-text">
+          ${this.verified}
+          <slot></slot>
```
Comment (atruskie) [inline src/components/decision/verification/verification.ts:104]: same as other - move text into slot

File: src/components/verification-grid/css/style.css [hunk 7]
```diff
@@ -15,6 +15,11 @@
   pointer-events: none;
 }
 
+#skip-button {
+  height: min-content;
```
Comment (atruskie) [inline src/components/verification-grid/css/style.css:19]: yeah not sure on this

File: src/components/verification-grid/verification-grid.ts [hunk 8]
```diff
@@ -324,8 +331,10 @@ export class VerificationGridComponent extends AbstractComponent(LitElement) {
 
     const decisionElements = this.decisionElements;
     if (decisionElements) {
-      decisionElements.forEach((element: DecisionComponent, i: number) => {
-        element.decisionId = i;
+      let nextDecisionId = 0;
+      decisionElements.forEach((element: DecisionComponent) => {
+        element.decisionId = nextDecisionId;
+        nextDecisionId += element.reservedDecisionIds;
```
Comment (atruskie) [inline src/components/verification-grid/verification-grid.ts:337]: needed?

**Review by atruskie — APPROVED** [review]
I don't really like that the skip button is a different size to other ones. All the decisions should have the same height?

I was thinking of showing <kbd>⇧</kbd>+<kbd>k</kbd> for a negative case. I think as implemented, noticing the case difference might be too subtle? It does looks neater though. Maybe this is something we test/get another opinion on?

The injector color service works well. Well done.

Minor changes suggested, but we'll talk offline. Get a style check with @nellihol  otherwise  this is great work.

File: src/components/decision/css/style.css (hunk 3 above)
Comment (hudson-newey) [reply src/components/decision/css/style.css:33]: Discussed offline, padding has been removed and replaced with height

File: src/components/decision/classification/classification.ts [hunk 9]
```diff
@@ -1,28 +1,185 @@
...
+
+  @query("#true-decision-button")
+  private trueDecisionButton!: HTMLButtonElement;
+
+  @query("#false-decision-button")
+  private falseDecisionButton!: HTMLButtonElement;
+
+  // for each classification decision component, we need to reserve two decision
+  // ids. One for the true decision and one for the false decision.
+  public reservedDecisionIds = 2 as const;
```
[the remark this answers is not in the document]
Comment (hudson-newey) [reply src/components/decision/classification/classification.ts:56]: Unfortunately, the linter does not enforce any sort of order.

I have a convention that I outlined in `code-style.md`, but it's not enforced

[the remark this answers is not in the document]
Comment (hudson-newey) [reply src/components/decision/classification/classification.ts:56]: We discussed offline, and decision ids are no longer needed

File: src/components/verification-grid/verification-grid.ts (hunk 8 above)
Comment (hudson-newey) [reply src/components/verification-grid/verification-grid.ts:337]: Removed decision ids

File: src/components/verification-grid/css/style.css (hunk 7 above)
Comment (hudson-newey) [reply src/components/verification-grid/css/style.css:19]: Mock-up without the min-content height

image

This is how it originally looked, but I (personally) decided that it looks bad

(Ignore the un-evenly sized buttons I'm trying out something from a previous comment, but haven't finished yet)

File: src/components/decision/classification/classification.ts (hunk 1 above)
Comment (hudson-newey) [reply src/components/decision/classification/classification.ts:155]: I don't quite understand what this is asking for

We do support disabling any element (including any buttons) through the `disabled` class in our themeing.css file

**Pull request merged**